x-pack/plugins/code/public/sagas/file.ts

```diff
@@ -54,6 +55,8 @@ function* handleFetchRepoTree(action: Action<FetchRepoTreePayload>) {
         // do not request file tree if this tree exists and its children are not empty
         if (!children || children.length === 0) {
           yield call(fetchPath, { uri, revision, path, parents, isDir });
+        } else {
```

Comment:
Also, this function is getting complicated and redundant now, there are multiple `yield call(fetchPath, { uri, revision, path, parents, isDir });` in the `if...else...`. This will make it easy to break

let's do some refactoring here, like

```
if (shouldFetchPath(action.payload)) {
   yield call(fetchpath, xxxx);
}
```